node/router/src/heartbeat.rs

```diff
@@ -140,13 +140,18 @@ pub trait Heartbeat<N: Network>: Outbound<N> {
     /// TODO (howardwu): If the node is a validator, keep the validator.
     /// This function keeps the number of connected peers within the allowed range.
     fn handle_connected_peers(&self) {
+        // Initialize an RNG.
+        let rng = &mut OsRng;
+
         // Obtain the number of connected peers.
         let num_connected = self.router().number_of_connected_peers();
         // Obtain the number of connected provers.
         let num_connected_provers = self.router().number_of_connected_provers();
 
+        // Consider rotating more external peers every ~10 heartbeats.
+        let reduce_peers = rng.gen_range(0..10) == 0 && self.router().rotate_external_peers();
```

Comment:
```suggestion
        let reduce_peers = self.router().rotate_external_peers() && rng.gen_range(0..10) == 0;
```
that way we avoid reaching into the OS randomness if the setting is disabled; it's a tiny perf difference, but there are no downsides